src/mod_websockets.erl

```diff
@@ -358,3 +358,38 @@ maybe_send_ping_request(none) ->
     ok;
 maybe_send_ping_request(PingRate) ->
     send_ping_request(PingRate).
+
+
+%%--------------------------------------------------------------------
+%% add_sec_websocket_protocol_header
+%%--------------------------------------------------------------------
+
+add_sec_websocket_protocol_header(Req) ->
+     case cowboy_req:header(<<"sec-websocket-protocol">>, Req) of
+         undefined ->
+             %% Server should not set "sec-websocket-protocol" respond header
+             %% if it does not present in the request
+             Req;
+         Proto ->
+             %% A client can propose several protocols
+             %% but we are interested in xmpp only
+             Protocols = split_protocol(Proto),
```

Comment:
There is `cowboy_req:parse_header/3` function which seems to be doing what you need here. Did you try it @arcusfelis?

Reply:
Good catch